File: frontend/src/types/account.ts
export type AccountStatus = 'pending' | 'approved' | 'in_copy' | 'expired' | 'suspended'
export type PhaseStatus = 'not_started' | 'in_progress' | 'passed' | 'failed'

export interface Account {
  id: number
  account_number: string
  account_password: string
  server: string
  buyer_name: string
  buyer_email: string | null
  buyer_phone: string | null
  buyer_notes: string | null
  purchase_date: string
  expiry_date: string | null
  purchase_price: number | null
  status: AccountStatus
  copy_count: number
  max_copies: number
  margin_size: number | null
  phase1_target: number | null
  phase1_status: PhaseStatus | null
  phase2_target: number | null
  phase2_status: PhaseStatus | null
  created_at: string
  updated_at: string | null
  created_by: number | null
}

export interface AccountCreate {
  account_number: string
  account_password: string
  server: string
  buyer_name: string
  buyer_email?: string
  buyer_phone?: string
  buyer_notes?: string
  purchase_date: string
  expiry_date?: string
  purchase_price?: number
  status?: AccountStatus
  max_copies?: number
  margin_size?: number
  phase1_target?: number
  phase1_status?: PhaseStatus
  phase2_target?: number
  phase2_status?: PhaseStatus
}

export interface AccountUpdate extends Partial<AccountCreate> {
  copy_count?: number
}

export interface Stats {
  total_accounts: number
  pending: number
  approved: number
  in_copy: number
  expired: number
  suspended: number
}

export interface AdminStats extends Stats {
  total_revenue: number
  accounts_this_month: number
}

File: backend/app/schemas/account.py
from pydantic import BaseModel, ConfigDict, EmailStr, Field
from datetime import date, datetime
from typing import Optional
from decimal import Decimal


class AccountBase(BaseModel):
    account_number: str
    server: str
    buyer_name: str
    buyer_email: Optional[EmailStr] = None
    buyer_phone: Optional[str] = None
    buyer_notes: Optional[str] = None
    purchase_date: date
    expiry_date: Optional[date] = None
    purchase_price: Optional[Decimal] = None
    status: str = "pending"
    max_copies: int = 1
    margin_size: Optional[Decimal] = None
    phase1_target: Optional[Decimal] = None
    phase1_status: Optional[str] = "not_started"
    phase2_target: Optional[Decimal] = None
    phase2_status: Optional[str] = None


class AccountCreate(AccountBase):
    account_password: str = Field(min_length=1)


class AccountUpdate(BaseModel):
    account_number: Optional[str] = None
    account_password: Optional[str] = None
    server: Optional[str] = None
    buyer_name: Optional[str] = None
    buyer_email: Optional[EmailStr] = None
    buyer_phone: Optional[str] = None
    buyer_notes: Optional[str] = None
    purchase_date: Optional[date] = None
    expiry_date: Optional[date] = None
    purchase_price: Optional[Decimal] = None
    status: Optional[str] = None
    copy_count: Optional[int] = None
    max_copies: Optional[int] = None
    margin_size: Optional[Decimal] = None
    phase1_target: Optional[Decimal] = None
    phase1_status: Optional[str] = None
    phase2_target: Optional[Decimal] = None
    phase2_status: Optional[str] = None


class AccountUpdateV2(BaseModel):
    account_number: Optional[str] = None
    server: Optional[str] = None
    buyer_name: Optional[str] = None
    buyer_email: Optional[EmailStr] = None
    buyer_phone: Optional[str] = None
    buyer_notes: Optional[str] = None
    purchase_date: Optional[date] = None
    expiry_date: Optional[date] = None
    purchase_price: Optional[Decimal] = None
    status: Optional[str] = None
    copy_count: Optional[int] = None
    max_copies: Optional[int] = None
    margin_size: Optional[Decimal] = None
    phase1_target: Optional[Decimal] = None
    phase1_status: Optional[str] = None
    phase2_target: Optional[Decimal] = None
    phase2_status: Optional[str] = None


class StatusUpdate(BaseModel):
    status: str


# Response for admin (all data)
class AccountAdminResponse(AccountBase):
    id: int
    account_password: str  # Decrypted for admin view
    copy_count: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    created_by: Optional[int] = None

    model_config = ConfigDict(from_attributes=True)


class AccountAdminV2Response(AccountBase):
    id: int
    copy_count: int
    created_at: datetime
    updated_at: Optional[datetime] = None
    created_by: Optional[int] = None

    model_config = ConfigDict(from_attributes=True)


# Response for public (filtered data)
class AccountPublicResponse(BaseModel):
    id: int
    status: str
    server: str
    copy_count: int
    max_copies: int
    is_available: bool

    model_config = ConfigDict(from_attributes=True)


# Stats response
class StatsResponse(BaseModel):
    total_accounts: int
    pending: int
    approved: int
    in_copy: int
    expired: int
    suspended: int


class AdminStatsResponse(StatsResponse):
    total_revenue: Decimal
    accounts_this_month: int


class PasswordRevealRequest(BaseModel):
    admin_password: str = Field(min_length=1)


class PasswordRevealResponse(BaseModel):
    account_password: str
    revealed_at: datetime
    expires_in_seconds: int


class PasswordRotateRequest(BaseModel):
    new_password: str = Field(min_length=8)
